File: 0078-subsets/0078-subsets.java
class Solution {
    private void backtrack(int start, List<Integer> curr, int[] nums, List<List<Integer>> res) {
        res.add(new ArrayList<>(curr));

        for(int i = start; i < nums.length; i++) {
            curr.add(nums[i]);
            backtrack(i + 1, curr, nums, res);
            curr.remove(curr.size() - 1);
        }
    }

    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> curr = new ArrayList<>();
        backtrack(0, curr, nums, res);
        return res;
    }
}

File: 0078-subsets/0078-subsets.go
func subsets(nums []int) [][]int {
    n := len(nums) 
    subsetCount := 1 << n
    var res [][]int

    for i := 0; i < subsetCount; i++ {
        var ds []int;
        for j := 0; j < n; j++ {
            if(i & (1 << j) != 0) {
                ds = append(ds, nums[j])
            }
        }
        res = append(res, ds);
    }

    return res;
}